torch/csrc/jit/codegen/cuda/mutator.cpp

```diff
@@ -80,27 +80,7 @@ Statement* OptOutMutator::mutate(TensorView* tv) {
 }
 
 Statement* OptOutMutator::mutate(kir::TensorIndex* ti) {
-  std::vector<Statement*> inds;
-  for (auto* ind : ti->indices())
-    inds.push_back(mutateAsVal(ind));
-
-  bool changed = false;
-  for (decltype(inds.size()) i{0}; i < inds.size(); i++) {
-    TORCH_INTERNAL_ASSERT(inds[i]->isVal() && inds[i]->asVal()->isAnInt());
-    if (!inds[i]->sameAs(ti->index(i)))
-      changed = true;
-  }
-
-  if (!changed)
-    return ti;
-
-  std::vector<Val*> valInds(inds.size(), nullptr);
-  for (decltype(inds.size()) i{0}; i < inds.size(); i++)
-    valInds[i] = inds[i]->asVal();
-
-  Val* mutated_val = new kir::TensorIndex(ti->view(), valInds);
-  registerMutation(ti, mutated_val);
-  return mutated_val;
```

Comment:
Why can this be removed?

Reply:
Doesn't seem to be needed. We may be able to eliminate the need for OptOutMutator completely after we refactor the lowering code.

Reply:
Why are only the bodies of the overloads for the kernel IR nodes removed? Wouldn't the class become inconsistent? Wouldn't it make more sense to remove the class entirely if it's not needed?

Reply:
That would be the end goal, although I believe that removing the class completely would require additional changes. 

I've been chipping away slowly, to get to the current state where we can have some confidence that it can be removed (when I started this was not as clear)

I opened #276 to track this.

Reply:
Sounds good to me.